### ISSUE TITLE
Feature: Back propagation Support

### ISSUE THREAD

**Issue by yashpatil333** [yashpatil333/p5-Neural-Network#1]
At the moment, it is only a feed forward Network! In the future, I would like to add support for Backward propagation of the net using training & test sample data using a Gradient Descent Algorithm. 

"Backpropagation is a supervised learning algorithm, for training Multi-layer Perceptrons"

Use case: Training Classification & Regression Models.
Example Tests: Mnist database for digit recognition.
